### PR TITLE
 Extract reference for operator FLOOR_DIV to standalone header

### PR DESCRIPTION
**Pull request by ddavis-2015**
Move the reference implementation to its own header so that micro
can use it without the unrelated depedencies of reference_ops.h.

This PR is part of the work to port operator FLOOR_DIV from lite to micro,
as tracked in Issue #45657.

### CODE REVIEW

**Review by petewarden — APPROVED** [review]

File: tensorflow/lite/kernels/internal/reference/floor_div.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_FLOOR_DIV_H_
+#define TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_FLOOR_DIV_H_
+
```
Comment (petewarden) [inline tensorflow/lite/kernels/internal/reference/floor_div.h:17]: Can you add #include <cmath> here? The build is failing internally with

```
third_party/tensorflow/lite/kernels/internal/reference/floor_div.h:27:15: error: no member named 'floor' in namespace 'std'
  return std::floor(std::divides<double>()(static_cast<double>(input1),
         ~~~~~^
1 error generated.
```

**Comment by gbaned** [comment]
@ddavis-2015 Can you please check @petewarden's comments and keep us posted ? Thanks!

File: tensorflow/lite/kernels/internal/reference/floor_div.h (hunk 1 above)
Comment (ddavis-2015) [reply tensorflow/lite/kernels/internal/reference/floor_div.h:17]: Additional include file now added.

**Comment by gbaned** [comment]
@ddavis-2015  Can you please resolve conflicts? Thanks!

**Review by petewarden — APPROVED** [review]

**Review by advaitjain — APPROVED** [review]

**Review by petewarden — APPROVED** [review]

**Pull request merged**